### ISSUE TITLE
Finish native libraries.

### ISSUE THREAD

**Issue by Ashvin-Ranjan** [nbuilding/N-lang#118]
Many of our native libraries are not finished and are just left with one or two functions, and some test libraries are still out there like `fek`.

Things to do:
- [x] ~~Remove `fek`~~
- [x] Allow for bytes in `FileIO`
- [x] `json`
  - [x] Formatting Options
  - [x] Convert Nan and infinities to null
- [x] `request`
  - [x] Add Different types of requests
    - [x] `HEAD`
    - [x] `PUT`
    - [x] `DELETE`
    - [x] `CONNECT`
    - [x] `OPTIONS`
    - [x] `TRACE`
    - [x] `PATCH`
  - [x] Add `http` servers (as discussed in #147)
- [x] Allow for console commands and `dll` use `SystemIO`
- [x] `times`
  - [x] `getTime`
  - [x]  date formatting
- [x] `websocket`
  - [x] `setInterval`
  - [x] Websocket debugging in Command Prompt
  - [x] Websocket server hosting (as discussed in #147)

**Comment by SheepTester** [comment]
We shouldn't remove fek for backwards compatibility reasons. I also see no reason why it can't stay

**Comment by SheepTester** [comment]
For making HTTP requests, we could just have a generic function that lets you set the request method, kind of like JavaScript `fetch`.

**Comment by Ashvin-Ranjan** [comment]
> We shouldn't remove fek for backwards compatibility reasons. I also see no reason why it can't stay

Fair enough, will remove

**Comment by Ashvin-Ranjan** [comment]
> For making HTTP requests, we could just have a generic function that lets you set the request method, kind of like JavaScript `fetch`.

I agree

**Comment by SheepTester** [comment]
Maybe for things like FileIO, SystemIO, websockets, etc., we can allow sending binary data using an enum that could either be binary data (a byte array) or text. Written in N, something like

```ts
// `text` assumes UTF-8; maybe we could have a separate variant that specifies the encoding?
type data = <binary list[byte]> | <text str>
```

Then we can send data either as text or binary data, like

```ts
FileIO.writeFile("./hello", text("Hello! 🐑"))!
FileIO.writeFile("./hello2", binary([1, 2, 3, 4]))!
```

We could introduce a new native type for bytes or byteArrays (it may be more efficient to make `byteArray` a native type as opposed to `list[byte]` at the cost of not having list methods available for byteArrays)

**Comment by SheepTester** [comment]
Also I think it might be a good idea for security reasons to require explicitly enabling permissions (perhaps through a command line argument). That way, if you run N code that uses a library, you can make sure none of your libraries can secretly access the command line if you don't expect them to.

For example, we could require `--allow-command-line` in order for anything that access the command line to be allowed to run; otherwise either we could return an `err` when something tries to access the command line or raise a type error

**Comment by Ashvin-Ranjan** [comment]
Seems like good.

**Comment by Ashvin-Ranjan** [comment]
Closed via #149
